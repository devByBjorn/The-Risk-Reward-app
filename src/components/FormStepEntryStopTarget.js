import React, { useState } from 'react'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'
import FormHelperText from '@material-ui/core/FormHelperText'
import FormPageContainer from '../components_style/FormPageContainerStyled'
import FormContainer from '../components_style/FormContainerStyled'
import FormNav from './FormNav'
import formElementsStyled from '../components_style/formElementsStyled'

const FormStepEntryStopTarget = ({
  values,
  nextStep,
  navigateByStepValue,
  prevStep,
  onChangeValue,
}) => {

  const classes = formElementsStyled()
  const { entry, stop, target, direction } = values

  const [error, setError] = useState(false)
  const [stopInfo, setStopInfo] = useState(' ')
  const [targetInfo, setTargetInfo] = useState(' ')

  const stopLogicLong = parseFloat(stop) <= parseFloat(entry)
  const stopLogicShort = parseFloat(stop) >= parseFloat(entry)
  const targetLogicLong = parseFloat(entry) <= parseFloat(target)
  const targetLogicShort = parseFloat(entry) >= parseFloat(target)

  const next = e => {
    e.preventDefault()
    if (!entry || !stop || !target) {
      setError(true)
    } else if (stopLogicShort && direction === 'long') {
      setStopInfo('Stop must be below entry when trade direction is long')
    } else if (stopLogicLong && direction === 'short') {
      setStopInfo('Stop must be above entry when trade direction is short')
    } else if (targetLogicShort && direction === 'long') {
      setTargetInfo('Target must be above entry when trade direction is long')
    } else if (targetLogicLong && direction === 'short') {
      setTargetInfo('Target must be below entry when trade direction is short')
    } else {
      setError(false)
      nextStep()
    }
  }

  const back = e => {
    e.preventDefault()
    prevStep()
  }

  return (
    <FormPageContainer>
      <FormNav
        values={values}
        navigateByStepValue={navigateByStepValue}
      />
      <FormContainer>
        <TextField
          className={classes.textField}
          error={!entry && error}
          label="Entry"
          name="entry"
          value={entry}
          onChange={onChangeValue}
          placeholder="Entry"
        />

        <TextField
          className={classes.textField}
          error={!stop && error}
          label="Stop"
          name="stop"
          value={stop}
          onChange={onChangeValue}
          placeholder="Stop"
        />
        <FormHelperText
          error={false}
          className={classes.helperText}
          children={stopInfo}
        />

        <TextField
          className={classes.textField}
          error={!target && error}
          label="Target"
          name="target"
          value={target}
          onChange={onChangeValue}
          placeholder="Target"
        />
        <FormHelperText
          className={classes.helperText}
          children={targetInfo}
        />
        <div className={classes.buttonContainer}>
          <Button
            className={classes.button}
            onClick={back}
          >Back</Button>
          <Button
            className={classes.button}
            onClick={next}
          >Next</Button>
        </div>
      </FormContainer>
    </FormPageContainer>
  )
}

export default FormStepEntryStopTarget