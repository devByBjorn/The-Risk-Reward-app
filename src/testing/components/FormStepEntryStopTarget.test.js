import React from 'react'
import { shallow } from 'enzyme'
import FormStepEntryStopTarget from '../../components/FormStepEntryStopTarget'
import { trades } from '../fixtures/trades'


test('should render FormParent', () => {
  const defaultValues = {
    direction: 'long',
    market: 'dax',
    entry: '',
    stop: '',
    target: '',
    status: 'closed',
    setup: 'ibl',
  }

  const wrapper = shallow(<FormStepEntryStopTarget values={defaultValues} />)
  expect(wrapper).toMatchSnapshot()
})

// test('should render error when no entry value', () => {
//   const nextStep = jest.fn()
//   const values = {
//     entry: '',
//     stop: '255',
//     target: '300',
//   }
//   const wrapper = shallow(<FormStepEntryStopTarget values={values} nextStep={nextStep} />)
//   wrapper.find('WithStyles(ForwardRef(Button))').at(1).simulate('click', {
//     preventDefault: () => { }
//   })

// })