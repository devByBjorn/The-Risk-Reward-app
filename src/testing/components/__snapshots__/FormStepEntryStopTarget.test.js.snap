// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormParent 1`] = `
<FormPageContainerStyled__FormPageContainer>
  <FormNav
    values={
      Object {
        "direction": "long",
        "entry": "",
        "market": "dax",
        "setup": "ibl",
        "status": "closed",
        "stop": "",
        "target": "",
      }
    }
  />
  <FormContainerStyled__FormContainer>
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-9"
      error={false}
      label="Entry"
      name="entry"
      placeholder="Entry"
      value=""
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-9"
      error={false}
      label="Initial stop"
      name="stop"
      placeholder="Initial stop"
      value=""
    />
    <WithStyles(ForwardRef(FormHelperText))
      className="makeStyles-helperText-6"
      error={false}
    >
       
    </WithStyles(ForwardRef(FormHelperText))>
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-9"
      label="Initial target"
      name="target"
      placeholder="Initial target"
      value=""
    />
    <WithStyles(ForwardRef(FormHelperText))
      className="makeStyles-helperText-6"
    >
       
    </WithStyles(ForwardRef(FormHelperText))>
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-9"
      label="Actual exit"
      name="exit"
      placeholder="Actual exit"
    />
    <div
      className="makeStyles-buttonContainer-1"
    >
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-2"
        onClick={[Function]}
      >
        Back
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-2"
        onClick={[Function]}
      >
        Next
      </WithStyles(ForwardRef(Button))>
    </div>
  </FormContainerStyled__FormContainer>
</FormPageContainerStyled__FormPageContainer>
`;
